### ISSUE TITLE
Docker::Error::ArgumentError: Must have id, got: {"id"=>nil, :headers=>{}}

### ISSUE THREAD

**Issue by marc-** [chef/chef-provisioning-docker#60]
Steps to reproduce:
1. Create image with `machine_image` resource;
2. Remove image created on previous step using `docker rmi` command;
3. Converge machine using image from step 1 as from_image attribute;

This happens due to chef-server has information regarding to specified image in `data bag`, but image is not available on target machine.

Error message doesn't clarify what has happened. We need to process errors like this and display more informative text.

Another option here, we can try to recreate images, since data bag item container information regarding to image specification. Though, it is currently blocked by #53 , image spec doesn't contain valid information about base image:
```json
{
  "id": "my_app",
  "reference": {
    "driver_url": "docker:unix:///var/run/docker.sock",
    "driver_version": "0.7-dev",
    "allocated_at": 1436044732,
    "docker_options": {
      "base_image": {
        "name": "chef_my_app",
        "repository": "chef",
        "tag": "my_app"
      },
      "from_image": true
    }
  },
  "run_list": [
    "recipe[test::recipe1]",
    "recipe[test::recipe2]",
    "recipe[test::recipe1]",
    "recipe[test::recipe2]"
  ]
}
```

**Comment by yazgoo** [comment]
From what I understand, this error basically means that the base image specified was not found ?
This is really confusing, plus modifying it in the recipe has no effect once the node was created (so you have to delete it first).

**Comment by bhavyaaha** [comment]
@marc- did you find any resolution to this ?

**Comment by marc-** [comment]
Nope, no resolution. From my perspective images has to be recreated.
